pkg/virt-launcher/virtwrap/sriov/hostdev.go

```diff
@@ -98,3 +102,82 @@ func decoratePciAddressField(addressField string) (*api.Address, error) {
 	}
 	return decoratedAddrField, nil
 }
+
+type deviceDetacher interface {
+	DetachDevice(xmlData string) error
+}
+
+func DetachHostDevices(dom deviceDetacher, hostDevices []api.HostDevice) error {
+	for _, device := range hostDevices {
+		devXML, err := xml.Marshal(device)
+		if err != nil {
+			return fmt.Errorf("failed to encode (xml) device %v, err: %v", device, err)
+		}
+		err = dom.DetachDevice(string(devXML))
+		if err != nil {
+			return fmt.Errorf("failed to detach device %s, err: %v", devXML, err)
+		}
+		log.Log.Infof("Live migration, hot-unplug device: %s (%v)", device.Alias.Name, device.Source.Address)
+	}
+	return nil
+}
+
+func FilterHostDevices(domainSpec *api.DomainSpec) []api.HostDevice {
```

Comment:
I think `GetHostDevices` / `ExtractHostDevices` will be a better fit, 
as far as I know `Filter..` functions return a collection without the elements you want to filter 
but his function does the opposite.
What you think?